cirq/study/sweeps.py

```diff
@@ -101,6 +102,38 @@ def __iter__(self) -> Iterator[resolver.ParamResolver]:
         for params in self.param_tuples():
             yield resolver.ParamResolver(collections.OrderedDict(params))
 
+    # pylint: disable=function-redefined
+    @overload
+    def __getitem__(self, val: int) -> resolver.ParamResolver:
+        pass
+
+    # pylint: disable=function-redefined
+    @overload
+    def __getitem__(self, val: slice) -> 'Sweep':
+        pass
+
+    def __getitem__(self, val):
+        if isinstance(val, int):
+            if val < 0:
+                val += len(self)
+            val = slice(val, val + 1, None)
+        if not isinstance(val, slice):
+            raise TypeError(
+                'Sweep indices must be either int or slices, not {}'\
+                    .format(type(val)))
+
+        inds_map: Dict[int, int] = {
+            val: i for i, val in enumerate(range(len(self))[val])
```

Comment:
use a different name for the loop variable, maybe do `i` -> `slice_i`, `val` -> `sweep_i`. Something like that.

Reply:
done.